File: Assignment3/4.c
#include<stdio.h>
int main(){
    int rows,cols;
    printf("Enter number of rows and columns: ");
    scanf("%d %d", &rows, &cols);
    int matrix[rows][cols];
    printf("Enter %d elements: ",rows*cols);
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            scanf("%d", &matrix[i][j]);
        }
    }
    int zero_count = 0;
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            if (matrix[i][j] == 0)
                zero_count++;
        }
    }

    if (zero_count > (rows * cols) / 2)
        printf("Matrix is Sparse\n");
    else
        printf("Matrix is Not Sparse\n");

    return 0;
}

File: Week-3/4.py
rows,cols = map(int(input("Enter rows and columns: ").split()))
matrix = []
for i in range(rows):
    row = int(input("Enter matrix elements row-wise: "))
    matrix.append(row)
zero_count = sum(row.count(0) for row in matrix)
if(zero_count > (rows * cols)//2):
    print("Matrix is sparse.")

else:
    print("Matrix is not sparse.")
